### PR TITLE
Adam/apirewrite to dev

### PR DESCRIPTION
**Pull request by labseven**
Made an actual RESTful api and used it in the webapps

### CODE REVIEW

File: static/js/editbob.js [hunk 1]
```diff
@@ -98,14 +105,14 @@ $(function() {
...
-		/**
-		 * Posts input data to the server using POST request
-		 * @param {string} url - A string containing the URL(route) to POST request
-		 * @param {Object} data - An object that contains a new bob data
-		 * @param {successCallback} function - A callback called upon success
-		*/
-		$.post('/editbob', data, function(res) {
-			alert('Bob saved!');
+		// Send the bob through the api with a PUT request
+		$.ajax({
```
Comment (SeunginLyu) [inline static/js/editbob.js]: Add some documentation about why we used ajax -> "put request requires ajax"

File: models/flavor.js [hunk 2]
```diff
@@ -19,16 +19,21 @@ function saveFlavor(flavorData) {
 }
 
 
-function getFlavors(filter) {
+function getAllFlavors(filter) {
```
Comment (aidankmcl) [inline models/flavor.js]: I think if this is explicitly meant to grab all flavors then this shouldn't accept a filter but should pass an empty object to `.find({})`

Comment (aidankmcl) [reply models/flavor.js]: If we still want search functionality I think the old name works fine.

File: models/flavor.js [hunk 3]
```diff
@@ -19,16 +19,21 @@ function saveFlavor(flavorData) {
...
-Flavors.model      = FlavorModel;
-Flavors.saveFlavor = saveFlavor;
-Flavors.getFlavors = getFlavors;
+Flavors.model         = FlavorModel;
+Flavors.saveFlavor    = saveFlavor;
+Flavors.getAllFlavors = getAllFlavors;
+Flavors.getFlavor     = getFlavor;
 
-module.exports     = Flavors;
+module.exports        = Flavors;
```
Comment (aidankmcl) [inline models/flavor.js]: -*-s-*-e-*-x-*-y-*--*--*-s-*-p-*-a-*-c-*-i-*-n-*-g-*-
I'm into it

File: routes/api.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+      res.send(data);
+    }, function error(err) {
+      res.status(500).send(err);
+    });
+  }
+
+  // Updates an existing bob
+  function PUTbob(req, res) {
+    var bob = {
+      _id:			 db.ObjectId(req.body.id),
```
Comment (aidankmcl) [inline routes/api.js]: do we need to update _id?

File: static/js/editbob.js [hunk 5]
```diff
@@ -14,64 +18,67 @@ $(function() {
...
-			});
-
-			// fills the tag forms
-			$.each(bob.tags, function (i) {
-				$(':input[value="' + bob.tags[i] + '"]').attr('checked', true);
-			});
-	  });
-	}
+  $.get('/api/bobs/' + bobId, function (bob) {
+		cur_bob = bob;
```
Comment (aidankmcl) [inline static/js/editbob.js]: is it just github or is the spacing around here a little funky?

**Review by aidankmcl — CHANGES_REQUESTED** [review]
Looking good in general! Some nitpicks and clarification questions

File: static/js/editbob.js (hunk 5 above)
Comment (labseven) [reply static/js/editbob.js]: There were two spaces next to each other. I'll check if it displays better on github now

File: routes/api.js (hunk 4 above)
Comment (labseven) [reply routes/api.js]: _id is used in db.Bob.updateBob, but is not actually written. Same with bob.flavor.

I added comments to explain this

**Review by aidankmcl — APPROVED** [review]

**Review by aidankmcl — APPROVED** [review]
hurrah!

**Pull request merged**